Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,545 @@
+    proxyInfo = {'OK': False}
+    if not os.getenv('X509_USER_PROXY'):
+      proxyInfo = getProxyInfo(disableVOMS=True)
+      if proxyInfo['OK']:
+        os.environ['X509_USER_PROXY'] = proxyInfo['Value']['path']
+        gLogger.debug('X509_USER_PROXY not defined. Using %s' % proxyInfo['Value']['path'])
+    try:
+      self.client = Client()
+    except (CannotAuthenticate, MissingClientParameter) as err:
+      if not proxyInfo['OK']:
+        proxyInfo = getProxyInfo(disableVOMS=True)
+      if proxyInfo['OK']:
+        dn = proxyInfo['Value']['identity']
+        username = proxyInfo['Value']['username']
+        os.environ['RUCIO_ACCOUNT'] = username
+        gLogger.debug('Switching to account %s mapped to proxy %s' %(username, dn))
+
+    try:
+      self.client = Client()
+      self.scopes = self.client.list_scopes()
+      self.account = self.client.account
+    except Exception as err:
+      gLogger.error('Cannot instantiate RucioFileCatalog interface, error : %s' % str(err))
+
+
+  def __getDidsFromLfn(self, lfn):
+    if lfn.find(':') > -1:
+      scope, name = lfn.split(':')
+      return {'scope': scope, 'name': name}
+    else:
+      scope = get_scope(lfn, scopes=self.scopes, client=self.client)
+    return {'scope': scope, 'name': lfn}
+
+
+  @checkCatalogArguments
+  def getCompatibleMetadata(self, queryDict, path, credDict):
+    """ Get distinct metadata values compatible with the given already defined metadata
+    """
+    self.client = Client()
+    if path != '/':
+      result = self.exists(path)
+      if not result['OK']:
+        return result
+      if not result['Value']:
+        return S_ERROR('Path not found: %s' % path)
+    result = S_OK({})
+    return result
+
+
+  @checkCatalogArguments
+  def getReplicas(self, lfns, allStatus = False):
+    """ Returns replicas for an LFN or list of LFNs
+    """
+    result = {'OK': True, 'Value': {'Successful': {}, 'Failed': {}}}
+    lfnChunks = breakListIntoChunks( lfns, 1000 )
+
+    self.client = Client()
+    for lfnList in lfnChunks:
+      try:
+        DidList = [self.__getDidsFromLfn(lfn) for lfn in lfnList if lfn]
+        for rep in self.client.list_replicas(DidList):
+          if rep:
+            lfn = rep['name']
+            if self.convertUnicode:
+              lfn = str(lfn)
+            if lfn not in result['Value']['Successful']:
+              result['Value']['Successful'][lfn] = {}
+            for rse in rep['rses']:
+              if self.convertUnicode:
+                result['Value']['Successful'][lfn][str(rse)] = str(rep['rses'][rse][0])
+              else:
+                result['Value']['Successful'][lfn][rse] = rep['rses'][rse][0]
+          else:
+            for did in DidList:
+              result['Value']['Failed'][did['name']] = 'Error'
+      except Exception as err:
+        return S_ERROR(str(err))
+    return result
+
+
+  @checkCatalogArguments
+  def listDirectory(self, lfns, verbose = False):
```

Comment:
verbose unused

Reply:
this function has a lot of if, it would be good to split it up or put some documentation in between at least